agent/src/main.cpp

```diff
@@ -250,8 +250,10 @@ int main(int argc, char **argv)
     } catch (ConfigError &error) {
         log_error() << "Could not load configuration";
         return 1;
-    } catch (ReturnCode failure) {
-        log_error() << "Agent initialization failed";
+    } catch (SoftwareContainerError &error) {
+        // Currently only case where any other exception can occur is during Agent init
+        std::string errorMessage = "Agent initialization failed: " + std::string(error.what());
```

Comment:
In this case no need to store the errorMessage since we can't propagate anywhere.